### DISCUSSION TITLE
Custom serialization for `bytes` - something instead of a string?

### DISCUSSION THREAD

**Discussion by treykeown** [litestar-org/litestar#3867] [Q&A]
I think it would be cool to serialize `bytes` on the Python side, and automatically have that show up as a `Uint8Array` on the JavaScript side. It should be possible in JS with a custom reviver, but can't I figure out how to control serialization of `bytes`. I'm trying to get something like this:

```
b"\xCA\xFE" -> {"__type__": "bytes", "value": "yv4="}
```

Is it possible to control how `bytes` are serialized? Seems like the default `msgspec` behavior is used.

**Comment by vinicq** [comment]
Override the `bytes` encoder via `type_encoders` on the Litestar app.

```python
import base64
from litestar import Litestar

def encode_bytes(value: bytes) -> dict[str, str]:
    return {"__type__": "bytes", "value": base64.b64encode(value).decode("ascii")}

app = Litestar(
    route_handlers=[...],
    type_encoders={bytes: encode_bytes},
)
```

That gives you the wrapped shape exactly. msgspec calls the encoder when it hits a `bytes` value inside any returned structure.

If you want to read that shape back into `bytes` when it comes in as a request body, you'll need a matching decoder (custom Pydantic/msgspec model with a field validator, or a `type_decoders` entry), since the encoder only covers the outbound side.

One gotcha. If a handler returns raw `bytes` as the response body itself (not nested inside another structure), Litestar treats it as binary content and short-circuits the encoder. The encoder fires when `bytes` appears inside a returned dict, list, or model, which is exactly what you want for the JS `Uint8Array` reviver path.
